marlowe-playground-client/src/MainFrame/State.purs

```diff
@@ -434,11 +450,16 @@ handleAction s (NewProjectAction action@(NewProject.CreateProject lang)) = do
 handleAction s (NewProjectAction action) = toNewProject $ NewProject.handleAction s action
 
 handleAction s (DemosAction action@(Demos.LoadDemo lang (Demos.Demo key))) = do
-  for_ (Map.lookup key StaticData.demoFiles) \contents -> HaskellEditor.editorSetValue contents
-  for_ (Map.lookup key StaticData.demoFilesJS) \contents -> void $ query _jsEditorSlot unit (Monaco.SetText contents unit)
-  for_ (preview (ix key) (Map.fromFoldable StaticData.marloweContracts)) \contents -> do
-    Simulation.editorSetValue contents
-    void $ query _blocklySlot unit (Blockly.SetCode contents unit)
+  case lang of
+    Haskell -> for_ (Map.lookup key StaticData.demoFiles) \contents -> HaskellEditor.editorSetValue contents
+    Javascript -> for_ (Map.lookup key StaticData.demoFilesJS) \contents -> void $ query _jsEditorSlot unit (Monaco.SetText contents unit)
+    Marlowe -> do
+      for_ (preview (ix key) (Map.fromFoldable StaticData.marloweContracts)) \contents -> do
```

Comment:
OOI, why is `marloweContracts` an `Array` (rather than a `Map`)?

Reply:
historical artefact, will fix now